docs/spark-queries.md

```diff
@@ -346,6 +346,9 @@ SELECT * FROM prod.db.table.partitions;
 Note:
 For unpartitioned tables, the partitions table will contain only the record_count and file_count columns.
 
+Note2:
+The output of the above query might differ between having copy-on-write or merge-on-read strategies. E.g. delete files with MOR strategy aren't applyied when producing the content of the partitions metadata table. As a result if you have renamed a partition (by updating the value of a partition column) then you would see both the 'old' and the 'new' one until you do a rewrite of delete/data files.
```

Comment:
```suggestion
The output of the above query might differ between having copy-on-write or merge-on-read strategies. For example, delete files with merge-on-read strategy aren't applied when producing the content of the partitions metadata table. As a result, if the table has renamed the partitions (by updating the value of a partition column), then the query output would contain both the 'old' and the 'new' partition values until rewrite data files (with delete files) is executed. 
```

Reply:
I think it is better to avoid personal pronouns like `you` and `we`. Hence the above suggestion along with some minor typos fix and nits.

Reply:
I actually left a comment on the issue, now I am wondering why we don't use metadata only delete in this case, to go to a codepath where we delete the manifests entry rather than avoid writing delete files.  I wonder if its because of doing an update

That being said, there may be use-cases where we delete enough of the partition using delete files in separate commit, that results in this, but I am not sure if this specific use case should result in this by design and we should put it specifically.

Reply:
Actualy I checked with @aokolnychyi , it seems the expected behavior.  Delete From may choose metadata-only delete, but update does not, as it is kind of an edge case.


What do you guys think about something simpler?  Rather than mentioning specific use-case.
```
The partitions metadata table shows partitions with data files or delete files in the current snapshot.  However, delete files are not applied, and so in some cases partitions may be shown even though all their data rows are marked deleted by delete files.
```

Reply:
Thanks for the analysis @szehon-ho! I'm fine adding the proposed comment if there is consensus.

Reply:
@szehon-ho: 

SGTM 👍🏻

Reply:
I changed the comment. Thanks for the suggestion, @szehon-ho!